test/chainweb/pact/local_test.exs

```diff
@@ -0,0 +1,130 @@
+defmodule Kadena.Chainweb.Client.CannedLocalRequests do
+  @moduledoc false
+
+  alias Kadena.Chainweb.Error
+  alias Kadena.Test.Fixtures.Chainweb
+
+  def request(
+        :post,
+        "https://api.testnet.chainweb.com/chainweb/0.0/testnet04/chain/1/pact/api/v1/local",
+        _headers,
+        body,
+        _options
+      ) do
+    case String.contains?(body, "(coin.get-balance 'bad')") do
+      true ->
+        response =
+          Error.new(
+            {:chainweb,
+             %{
+               status: 400,
+               title: "Validation failed: Invalid command: Failed reading: empty"
+             }}
+          )
+
+        {:error, response}
+
+      false ->
+        response = Chainweb.fixture("local")
+        {:ok, response}
+    end
+  end
+end
+
+defmodule Kadena.Chainweb.Pact.LocalTest do
+  @moduledoc """
+  `Local` endpoint implementation tests.
+  """
+  use ExUnit.Case
+  alias Kadena.Chainweb.Client.CannedLocalRequests
+  alias Kadena.Chainweb.{Error, Pact}
+  alias Kadena.Chainweb.Pact.LocalResponse
+  alias Kadena.Cryptography
+  alias Kadena.Pact.ExecCommand
+  alias Kadena.Types.MetaData
+
+  setup do
+    Application.put_env(:kadena, :http_client_impl, CannedLocalRequests)
+
+    on_exit(fn ->
+      Application.delete_env(:kadena, :http_client_impl)
+    end)
+
+    code = "(+ 5 6)"
+    bad_code = "(coin.get-balance 'bad')"
+    cmd = create_command(code)
+    cmd_for_error = create_command(bad_code)
+
+    %{
+      cmd: cmd,
+      cmd_for_error: cmd_for_error,
+      local_response:
+        {:ok,
+         %LocalResponse{
+           continuation: nil,
+           events: nil,
+           gas: 6,
+           logs: "wsATyGqckuIvlm89hhd2j4t6RMkCrcwJe_oeCYr7Th8",
+           meta_data: %{
+             block_height: 3_303_861,
+             block_time: 1_671_546_034_831_940,
+             prev_block_hash: "Y6Wj0sxJpdV8M-3ihAbzUka57Wv5ZV2Uez6H_6_WeeY",
+             public_meta: %{
+               chain_id: "1",
+               creation_time: 1_671_462_208,
+               gas_limit: 1000,
+               gas_price: 1.0e-6,
+               sender: "k:d1a361d721cf81dbc21f676e6897f7e7a336671c0d5d25f87c10933cac6d8cf7",
+               ttl: 28_800
+             }
+           },
+           req_key: "SvVzKL5KC0r1JdIkJDhMHT8vZhmO2CdTxPKX-6YmGG0",
+           result: %{data: 11, status: "success"},
+           tx_id: nil
+         }}
+    }
```

Comment:
@EdwinGuayacan WDYT about this way to pass the params in the setup?

```elixir
%{
  success: %{cmd: cmd, response: response},
  error: %{cmd: cmd, response: response}
}
```

Reply:
looks better!